tests/_test_utils/test_auth.py

```diff
@@ -48,12 +48,16 @@ def test_keyring():
 
 
 def test_get_auth_from_url_and_name():
+    # set start condition to prevent definitions from a test earlier
+    constants.AUTH_LOGIN_CACHE == {}
     # empty http_auth definition
     server_url = "http://example.com"
     http_auth = config_loader(dataset="MSS_auth")
     assert http_auth == {}
     data = auth.get_auth_from_url_and_name(server_url, http_auth, overwrite_login_cache=False)
     assert data == (None, None)
+    # checking if the test setup changes this
+    assert constants.AUTH_LOGIN_CACHE == {}
```

Comment:
this test on github sometimes seems like to have a bias from another test, related to TestMigration. It happens only here in serial tests. And it does not happen on each run.   In that case at some point  constants.AUTH_LOGIN_CACHE is not empty.